src/archiefvernietigingscomponent/destruction/models.py

```diff
@@ -103,6 +103,7 @@ def next_assignee(self, review=None):
         return None
 
     def assign(self, assignee):
+
```

Comment:
The problem is actually here. The name chosen for this function argument was not ideal. Here `assignee` is actually a `User` (and not a `DestructionListAssignee`). I suggest that we  rename `assignee` in the function to be `assigned_user` and add a type hint (i.e. `def assign(self, assigned_user: User) -> None:`).

Then, we need to retrieve the actual `DestructionListAssignee` related to the `assigned_user` and the current `DestructionList` (self). Once you have retrieved the assignee, then you can set the `assigned_on` field.

Reply:
so you suggest I should add the assign_on to the assign in the model rather than the form?